app/preprints/detail/route.ts

```diff
@@ -80,6 +81,28 @@ export default class PreprintsDetail extends Route {
             const subjects = await preprint?.queryHasMany('subjects');
             const versions = await preprint?.queryHasMany('versions');
 
+            const preprintWithdrawableState = [ReviewsState.ACCEPTED, ReviewsState.PENDING]
+                .includes(preprint.reviewsState);
+            let isWithdrawalRejected = false;
+            let hasPendingWithdrawal = false;
+            if (preprintWithdrawableState) {
+                const withdrawalRequests = await preprint?.queryHasMany('requests');
+                const latestWithdrawalRequest = withdrawalRequests.firstObject;
+                if (latestWithdrawalRequest) {
+                    hasPendingWithdrawal = latestWithdrawalRequest.machineState === 'pending';
+                    const requestActions = await withdrawalRequests.firstObject?.queryHasMany('actions', {
```

Comment:
It's a shame this has to block rendering waiting to resolve.

Reply:
Yea, the component _could_ fetch the requests/actions and whatnot, but it might cause some of the content on the page to jump around as the requests get fulfilled... We _might_ be able to use the `include` option when getting the preprint, but I'm not sure how well-supported that function is in the API